src/extension.ts

```diff
@@ -47,7 +48,8 @@ async function getClusterKubeconfig(target: AksClusterTreeItem): Promise<string
         vscode.window.showErrorMessage(`Invalid ARM id ${target.id}`);
         return;
     }
-    const client = new azcs.ContainerServiceClient(target.root.credentials, target.root.subscriptionId);  // TODO: safely
+    const credentials = <msRestJs.ServiceClientCredentials> <unknown>target.root.credentials;
```

Comment:
Very minor style suggestion: tuck away the ugly type assertions in a named helper e.g. `restJSCredentialsFrom(...)`